functionary/core/models/workflow.py

```diff
@@ -0,0 +1,46 @@
+import uuid
+
+from django.conf import settings
+from django.db import models
+
+
+class Workflow(models.Model):
+    """A Workflow defines a series of tasks to be executed in sequence.
+
+    Attributes:
+        id: unique identifier (UUID)
+        environment: the environment that this task belongs to. All queryset filtering
+                     should include an environment.
+        name: the name of the Workflow
+        description: details about the Workflow
+        schema: describes the parameters that runs of this Workflow may provide
+        creator: the user that initiated the task
+        created_at: task creation timestamp
+        updated_at: task updated timestamp
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    environment = models.ForeignKey(to="Environment", on_delete=models.CASCADE)
+    name = models.CharField(max_length=64)
```

Comment:
Does this need a validator like the WorkflowStep's name?

Reply:
I don't think so, since the name here is just a friendly name mostly for human identification.  The `WorkflowStep` name has the restrictions because it has to be a legal name for use by the templating engine.